### ISSUE TITLE
Comparative Analysis For Spam Filtering (Fine Tuning and NLP Centric)

### ISSUE THREAD

**Issue by mderosa001** [mderosa001/Capstone_Issues#15]
### Title
Next-Generation Spam Filtering: Comparative Fine-Tuning of LLMs, NLPs, and CNN Models for Email Spam Classification
### URL
https://www.mdpi.com/2079-9292/13/11/2034#B5-electronics-13-02034
### Summary 
This research investigates the effectiveness of fine-tuning advanced language models, specifically GPT-4, BERT, and RoBERTa, for improved email spam and phishing attack detection, comparing their performance against traditional methods like Convolutional Neural Networks (CNNs). Through meticulous fine-tuning on two distinct email datasets and rigorous evaluation, including cross-dataset analysis to assess generalization, the study demonstrates that fine-tuned LLMs and NLP models significantly outperform CNNs, achieving nearly 100% accuracy in spam classification, with GPT-4 exhibiting superior generalization capabilities. The findings highlight the potential of leveraging these advanced language models to develop more robust and adaptive spam filtering systems, addressing the evolving challenges posed by sophisticated cyber threats and enhancing overall email security.
### Key Points 
Fine tuning LLM's and NLP models out preform CNN and are close to 100% accurate
### Citation
@Article{electronics13112034,
AUTHOR = {Roumeliotis, Konstantinos I. and Tselikas, Nikolaos D. and Nasiopoulos, Dimitrios K.},
TITLE = {Next-Generation Spam Filtering: Comparative Fine-Tuning of LLMs, NLPs, and CNN Models for Email Spam Classification},
JOURNAL = {Electronics},
VOLUME = {13},
YEAR = {2024},
NUMBER = {11},
ARTICLE-NUMBER = {2034},
URL = {https://www.mdpi.com/2079-9292/13/11/2034},
ISSN = {2079-9292},
ABSTRACT = {Spam emails and phishing attacks continue to pose significant challenges to email users worldwide, necessitating advanced techniques for their efficient detection and classification. In this paper, we address the persistent challenges of spam emails and phishing attacks by introducing a cutting-edge approach to email filtering. Our methodology revolves around harnessing the capabilities of advanced language models, particularly the state-of-the-art GPT-4 Large Language Model (LLM), along with BERT and RoBERTa Natural Language Processing (NLP) models. Through meticulous fine-tuning tailored for spam classification tasks, we aim to surpass the limitations of traditional spam detection systems, such as Convolutional Neural Networks (CNNs). Through an extensive literature review, experimentation, and evaluation, we demonstrate the effectiveness of our approach in accurately identifying spam and phishing emails while minimizing false positives. Our methodology showcases the potential of fine-tuning LLMs for specialized tasks like spam classification, offering enhanced protection against evolving spam and phishing attacks. This research contributes to the advancement of spam filtering techniques and lays the groundwork for robust email security systems in the face of increasingly sophisticated threats.},
DOI = {10.3390/electronics13112034}
}
